### PR TITLE
Fix FanGraphs Data Issues & Various Test Fixes

### LINKED ISSUE #194: pitching_stats not executing due to Fangraphs data changes
KeyError: 'Zone% (pfx)'

Looks like the addition of new stats onto Fangraphs also led to the removal of the 'pfx' stats

### PR DESCRIPTION
**Pull request by TheCleric**
* Update the cache type so that it actually populates the downstream func in VS Code
* Update the Fangraphs enums to support the new and renamed columns
* Add tests for Fangraphs enums to compare them against the ones listed on Fangraphs' site
* Fix a few minor pylint nits
* Due to the change of Fielding enums, adding the pull of fielding_stats like we do batting_stats and pitching_stats
* Fix `statcast_pitcher_active_spin` (assuming due to the fact that Statcast changed the endpoint)
* Fixed attrs not being a true requirement in setup.py
* Fix the Fangraphs data tests to not have hard coded column counts
* Fix some season tests that fail with current season when season hasn't started yet

Fixes #194 and possibly #193

### CODE REVIEW

File: pybaseball/datasources/fangraphs.py [hunk 1]
```diff
@@ -153,7 +151,7 @@ def fetch(self, start_season: int, end_season: Optional[int] = None, league: str
         }
 
         return self._validate(
-            self._postprocess( 
+            self._postprocess(
```
Comment (schorrm) [inline pybaseball/datasources/fangraphs.py:154]: _sigh_ we really do need an autoformatter
(you're 100% in the right to fix this, but avoiding the stupid minimal diffs would be great

File: tests/integration/pybaseball/datasources/test_fg_pitching_data.py [hunk 2]
```diff
@@ -120,15 +123,15 @@ def test_fg_pitching_maximum_age(self) -> None:
         assert len(data.columns) == self.ALL_DATA_COLUMNS_COUNT
         assert len(data.index) == 1
 
-    def test_fg_pitching_team(self, assert_frame_not_equal) -> None:
+    def test_fg_pitching_team(self, assert_frame_not_equal: Callable[[pd.DataFrame, pd.DataFrame], None]) -> None:
```
Comment (schorrm) [inline tests/integration/pybaseball/datasources/test_fg_pitching_data.py:126]: Wasn't there a specific thingy above for this?

File: tests/integration/pybaseball/enums/fangraphs/transforms.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        if leaderboard_item.endswith(_from):
+            leaderboard_item = leaderboard_item[:-1*len(_from)] + _to
+
+    for _from, _to in mid_replacements.items():
+        leaderboard_item = leaderboard_item.replace(_from, _to)
+
+    while '  ' in leaderboard_item:
+        leaderboard_item = leaderboard_item.replace('  ', ' ')
+
+    return leaderboard_item.strip().replace(' ', '_')
```
Comment (schorrm) [inline tests/integration/pybaseball/enums/fangraphs/transforms.py:49]: I don't understand what this is about

File: tests/integration/pybaseball/test_statcast_pitcher.py [hunk 4]
```diff
@@ -1,102 +1,102 @@
...
-	assert result is not None
-	assert not result.empty
+    assert result is not None
+    assert not result.empty
 
-	assert len(result.columns) == 30
-	assert len(result) == 260
\ No newline at end of file
+    assert len(result.columns) == 30
+    assert len(result) == 260
```
Comment (schorrm) [inline tests/integration/pybaseball/test_statcast_pitcher.py]: Is this necessary? (_sighs about formatter again_)

File: tests/integration/pybaseball/test_statcast_batter.py [hunk 5]
```diff
@@ -51,10 +51,11 @@ def test_statcast_batter_percentile_ranks() -> None:
...
-    result: pd.DataFrame = statcast_batter_pitch_arsenal(2019)
+    min_pa = 25
+    result: pd.DataFrame = statcast_batter_pitch_arsenal(2019, min_pa)
 
     assert result is not None
     assert not result.empty
 
     assert len(result.columns) == 21
-    assert len(result) == 2204
+    assert len(result) == 2205
```
Comment (schorrm) [inline tests/integration/pybaseball/test_statcast_batter.py:61]: Maybe loosen this assert like we did for other SC?

File: tests/integration/pybaseball/enums/fangraphs/transforms.py (hunk 3 above)
Comment (schorrm) [inline tests/integration/pybaseball/enums/fangraphs/transforms.py:49]: What's this about?

File: tests/integration/pybaseball/test_standings.py [hunk 6]
```diff
@@ -9,7 +9,7 @@
 
 def get_division_counts_by_season(season: Optional[int]) -> int:
     if season is None:
-        season = most_recent_season()
+        season = most_recent_season() - 1
```
Comment (schorrm) [inline tests/integration/pybaseball/test_standings.py:12]: Solid ✔️

File: tests/integration/pybaseball/test_standings.py [hunk 7]
```diff
@@ -19,7 +19,7 @@ def get_division_counts_by_season(season: Optional[int]) -> int:
 
 class TestBRefStandings:
     @pytest.mark.parametrize(
-        "season", [(x) for x in range(1871, most_recent_season() + 1)] + [(None)] # type: ignore
+        "season", [(x) for x in range(1871, most_recent_season() - 1)]
```
Comment (schorrm) [inline tests/integration/pybaseball/test_standings.py:22]: The `-1` here should be incorrect, no? Through `most_recent_season()` in the range = MRS - 1?

File: tests/integration/pybaseball/test_standings.py [hunk 8]
```diff
@@ -45,3 +45,10 @@ def test_standings_future(self) -> None:
         standings_list = standings(season)
 
         assert standings_list == []
+
+    def test_standings_none(self) -> None:
+        season = None
+
+        standings_list = standings(season)
+
+        assert standings_list == []
```
Comment (schorrm) [inline tests/integration/pybaseball/test_standings.py:54]: Can you explain the rationale here?

File: pybaseball/fielding_leaders.py [hunk 9]
```diff
@@ -0,0 +1,4 @@
+from .datasources.fangraphs import fg_fielding_data
+
+# This is just a pass through for the new, more configurable function
+fielding_stats = fg_fielding_data
```
Comment (schorrm) [inline pybaseball/fielding_leaders.py:4]: Do we need a file for this?

File: pybaseball/statcast_pitcher.py [hunk 10]
```diff
@@ -71,10 +70,31 @@ def statcast_pitcher_pitch_movement(year: int, minP: Union[int, str] = "q", pitc
...
+
+
+    url = f"https://baseballsavant.mlb.com/leaderboard/active-spin?year={year}_{_type}&min={minP}&hand=&csv=true"
     res = requests.get(url, timeout=None).content
+    if res and '<html' in res.decode('utf-8'):
+        # This did no go as planned. Statcast redirected us back to HTML :(
+        if _type == 'spin-based':
+            return statcast_pitcher_active_spin(year, minP, 'observed')
+        
+        return pd.DataFrame()
```
Comment (schorrm) [inline pybaseball/statcast_pitcher.py:92]: Should we warn something based on the comment there?

File: pybaseball/utils.py [hunk 11]
```diff
@@ -287,4 +287,4 @@ def norm_pitch_code(pitch: str, to_word: bool = False) -> str:
         if pitch.lower() is 'all':
             raise ValueError("'All' is not a valid pitch in this particular context!")
         raise ValueError(f'{pitch} is not a valid pitch!')
-    return normed
\ No newline at end of file
+    return normed
```
Comment (schorrm) [inline pybaseball/utils.py]: _sigh_

File: pybaseball/datasources/fangraphs.py (hunk 1 above)
Comment (TheCleric) [reply pybaseball/datasources/fangraphs.py:154]: To be fair, this wasn't the autoformatter as much as it was bugging me and I have a low grade case of OCD. 😆 

Does that make me the autoformatter?

File: tests/integration/pybaseball/datasources/test_fg_pitching_data.py (hunk 2 above)
Comment (TheCleric) [reply tests/integration/pybaseball/datasources/test_fg_pitching_data.py:126]: Not for this one, but I'll add one.

File: tests/integration/pybaseball/test_statcast_pitcher.py (hunk 4 above)
Comment (TheCleric) [reply tests/integration/pybaseball/test_statcast_pitcher.py]: Actually this was me again. Necessary? Not 100% (and I can revert it), I've just seen too many issues from having a mix of tabs and spaces in a project, especially with multiple participants working in multiple environments.

File: tests/integration/pybaseball/enums/fangraphs/transforms.py (hunk 3 above)
Comment (TheCleric) [reply tests/integration/pybaseball/enums/fangraphs/transforms.py:49]: I'll add better documentation strings

File: tests/integration/pybaseball/test_standings.py (hunk 7 above)
Comment (TheCleric) [reply tests/integration/pybaseball/test_standings.py:22]: Yep you're right, will fix.

File: tests/integration/pybaseball/test_standings.py (hunk 8 above)
Comment (TheCleric) [reply tests/integration/pybaseball/test_standings.py:54]: This was already part of the range test above, but was acting funky. Running it again today it's back to working. In theory the None should get replaced with a sane default, but between our func and the actual call, it's not behaving, so I'm removing it.

File: pybaseball/fielding_leaders.py (hunk 9 above)
Comment (TheCleric) [reply pybaseball/fielding_leaders.py:4]: Need? No. I was just trying to be consistent with the other two (batting_stats, pitching_stats).

File: pybaseball/statcast_pitcher.py (hunk 10 above)
Comment (TheCleric) [reply pybaseball/statcast_pitcher.py:92]: Probably not a bad idea.

File: pybaseball/utils.py (hunk 11 above)
Comment (TheCleric) [reply pybaseball/utils.py]: Tough, but fair.

**Comment by schorrm** [comment]
LGTM -- @wfordh, @TheCleric -- you guys approve?

**Pull request merged**